# Netflix

### Load and read data

In [2]:
# Import libraries
import pandas as pd
import numpy as np

In [3]:
# Read data file
df = pd.read_csv("netflix.csv")

### Data preprocessing

In [4]:
# Display rows and columns of data
df.shape

(8807, 12)

In [5]:
# Display data
df

show_id     type                  title         director  \
0         s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1         s2  TV Show          Blood & Water              NaN   
2         s3  TV Show              Ganglands  Julien Leclercq   
3         s4  TV Show  Jailbirds New Orleans              NaN   
4         s5  TV Show           Kota Factory              NaN   
...      ...      ...                    ...              ...   
8802   s8803    Movie                 Zodiac    David Fincher   
8803   s8804  TV Show            Zombie Dumb              NaN   
8804   s8805    Movie             Zombieland  Ruben Fleischer   
8805   s8806    Movie                   Zoom     Peter Hewitt   
8806   s8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                                   NaN  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                   NaN            NaN   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                                NaN            NaN   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020  PG-13     90 min   
1     September 24, 2021          2021  TV-MA  2 Seasons   
2     September 24, 2021          2021  TV-MA   1 Season   
3     September 24, 2021          2021  TV-MA   1 Season   
4     September 24, 2021          2021  TV-MA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018  TV-Y7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015  TV-14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

In [6]:
# Display basic statistical details
df.describe()

release_year
count   8807.000000
mean    2014.180198
std        8.819312
min     1925.000000
25%     2013.000000
50%     2017.000000
75%     2019.000000
max     2021.000000

In [7]:
# Print full summary
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


In [8]:
# Display columns
df.columns

Index(['show_id', 'type', 'title', 'director', 'cast', 'country', 'date_added',
       'release_year', 'rating', 'duration', 'listed_in', 'description'],
      dtype='object')

### Duplicates

In [9]:
# Check for duplicates
df.duplicated().sum()

0

In [9]:
# Check number of rows where particular columns of null values
# df.isna().sum()

df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

In [10]:
# % of missing data
total = np.product(df.shape)
total_null = df.isnull().sum().sum()
percentage_missing = (total_null/total)*100
percentage_missing

4.075356723818175

There are 4.07% missing data null values.

### Dropna

In [11]:
# Method 1
# Remove null values
df_new = df.dropna()
df_new

show_id     type                          title             director  \
7         s8    Movie                        Sankofa         Haile Gerima   
8         s9  TV Show  The Great British Baking Show      Andy Devonshire   
9        s10    Movie                   The Starling       Theodore Melfi   
12       s13    Movie                   Je Suis Karl  Christian Schwochow   
24       s25    Movie                          Jeans           S. Shankar   
...      ...      ...                            ...                  ...   
8801   s8802    Movie                        Zinzana      Majid Al Ansari   
8802   s8803    Movie                         Zodiac        David Fincher   
8804   s8805    Movie                     Zombieland      Ruben Fleischer   
8805   s8806    Movie                           Zoom         Peter Hewitt   
8806   s8807    Movie                         Zubaan          Mozez Singh   

                                                   cast  \
7     Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8     Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12    Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   
24    Prashanth, Aishwarya Rai Bachchan, Sri Lakshmi...   
...                                                 ...   
8801  Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                                country          date_added  \
7     United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                        United Kingdom  September 24, 2021   
9                                         United States  September 24, 2021   
12                              Germany, Czech Republic  September 23, 2021   
24                                                India  September 21, 2021   
...                                                 ...                 ...   
8801                       United Arab Emirates, Jordan       March 9, 2016   
8802                                      United States   November 20, 2019   
8804                                      United States    November 1, 2019   
8805                                      United States    January 11, 2020   
8806                                              India       March 2, 2019   

      release_year rating   duration  \
7             1993  TV-MA    125 min   
8             2021  TV-14  9 Seasons   
9             2021  PG-13    104 min   
12            2021  TV-MA    127 min   
24            1998  TV-14    166 min   
...            ...    ...        ...   
8801          2015  TV-MA     96 min   
8802          2007      R    158 min   
8804          2009      R     88 min   
8805          2006     PG     88 min   
8806          2015  TV-14    111 min   

                                             listed_in  \
7     Dramas, Independent Movies, International Movies   
8                         British TV Shows, Reality TV   
9                                     Comedies, Dramas   
12                        Dramas, International Movies   
24     Comedies, International Movies, Romantic Movies   
...                                                ...   
8801           Dramas, International Movies, Thrillers   
8802                    Cult Movies, Dramas, Thrillers   
8804                           Comedies, Horror Movies   
8805                Children & Family Movies, Comedies   
8806    Dramas, International Movies, Music & Musicals   

                                            description  
7     On a photo shoot in Ghana, an American model s...  
8     A talented batch of amateur bakers face off in...  
9     A woman adjusting to life after a loss contend...  
12    After 

In [12]:
# Check number of rows where particular columns of null values removed
df_new.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

In [13]:
# Method 1a
# Remove null values
df_new = df.copy()

# Inplace means reassign values to df_new after dropping columns
df_new.dropna(inplace = True)
df_new.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

In [14]:
# Display rows and columns of data after dropping null values
df_new.shape

(5332, 12)

It can be observed that the rows dataset has been decreased by 3475 from 8807 to 5332.

### Fillna

In [15]:
df_new1 = df.copy()
df_new1.isna().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

In [16]:
df_new1.fillna(0, inplace = True)
df_new1.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

> ffill

In [17]:
df_new_f = df.copy()

# axis = 0 (rows)
df_new_f.fillna(method = "ffill", axis = 0, inplace = True)
df_new_f.isna().sum()

show_id         0
type            0
title           0
director        0
cast            1
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

> bfill

In [18]:
df_new_b = df.copy()

# axis = 0 (rows)
df_new_b.fillna(method = "bfill", axis = 0, inplace = True)
df_new_b.isna().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

In [19]:
# Display data after dropping na
df_new

show_id     type                          title             director  \
7         s8    Movie                        Sankofa         Haile Gerima   
8         s9  TV Show  The Great British Baking Show      Andy Devonshire   
9        s10    Movie                   The Starling       Theodore Melfi   
12       s13    Movie                   Je Suis Karl  Christian Schwochow   
24       s25    Movie                          Jeans           S. Shankar   
...      ...      ...                            ...                  ...   
8801   s8802    Movie                        Zinzana      Majid Al Ansari   
8802   s8803    Movie                         Zodiac        David Fincher   
8804   s8805    Movie                     Zombieland      Ruben Fleischer   
8805   s8806    Movie                           Zoom         Peter Hewitt   
8806   s8807    Movie                         Zubaan          Mozez Singh   

                                                   cast  \
7     Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8     Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   
12    Luna Wedler, Jannis Niewöhner, Milan Peschel, ...   
24    Prashanth, Aishwarya Rai Bachchan, Sri Lakshmi...   
...                                                 ...   
8801  Ali Suliman, Saleh Bakri, Yasa, Ali Al-Jabri, ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...   

                                                country          date_added  \
7     United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                        United Kingdom  September 24, 2021   
9                                         United States  September 24, 2021   
12                              Germany, Czech Republic  September 23, 2021   
24                                                India  September 21, 2021   
...                                                 ...                 ...   
8801                       United Arab Emirates, Jordan       March 9, 2016   
8802                                      United States   November 20, 2019   
8804                                      United States    November 1, 2019   
8805                                      United States    January 11, 2020   
8806                                              India       March 2, 2019   

      release_year rating   duration  \
7             1993  TV-MA    125 min   
8             2021  TV-14  9 Seasons   
9             2021  PG-13    104 min   
12            2021  TV-MA    127 min   
24            1998  TV-14    166 min   
...            ...    ...        ...   
8801          2015  TV-MA     96 min   
8802          2007      R    158 min   
8804          2009      R     88 min   
8805          2006     PG     88 min   
8806          2015  TV-14    111 min   

                                             listed_in  \
7     Dramas, Independent Movies, International Movies   
8                         British TV Shows, Reality TV   
9                                     Comedies, Dramas   
12                        Dramas, International Movies   
24     Comedies, International Movies, Romantic Movies   
...                                                ...   
8801           Dramas, International Movies, Thrillers   
8802                    Cult Movies, Dramas, Thrillers   
8804                           Comedies, Horror Movies   
8805                Children & Family Movies, Comedies   
8806    Dramas, International Movies, Music & Musicals   

                                            description  
7     On a photo shoot in Ghana, an American model s...  
8     A talented batch of amateur bakers face off in...  
9     A woman adjusting to life after a loss contend...  
12    After 

### Replace na

In [20]:
# Check the number of null values under rating column
df[df['duration'].isna()]

show_id   type                                 title    director  \
5541   s5542  Movie                       Louis C.K. 2017  Louis C.K.   
5794   s5795  Movie                 Louis C.K.: Hilarious  Louis C.K.   
5813   s5814  Movie  Louis C.K.: Live at the Comedy Store  Louis C.K.   

            cast        country          date_added  release_year  rating  \
5541  Louis C.K.  United States       April 4, 2017          2017  74 min   
5794  Louis C.K.  United States  September 16, 2016          2010  84 min   
5813  Louis C.K.  United States     August 15, 2016          2015  66 min   

     duration listed_in                                        description  
5541      NaN    Movies  Louis C.K. muses on religion, eternal love, gi...  
5794      NaN    Movies  Emmy-winning comedy writer Louis C.K. brings h...  
5813      NaN    Movies  The comic puts his trademark hilarious/thought...

It can be observed that duration column has been filled with nan values, while rating column is filled with duration.

In [21]:
df.loc[df['show_id'] == 's5542', 'duration'] = '74 min'
df.loc[df['show_id'] == 's5795', 'duration'] = '84 min'
df.loc[df['show_id'] == 's5814', 'duration'] = '66 min'

#see the rows after replace action
df[df['show_id'].isin(['s5542','s5795','s5814'])]

show_id   type                                 title    director  \
5541   s5542  Movie                       Louis C.K. 2017  Louis C.K.   
5794   s5795  Movie                 Louis C.K.: Hilarious  Louis C.K.   
5813   s5814  Movie  Louis C.K.: Live at the Comedy Store  Louis C.K.   

            cast        country          date_added  release_year  rating  \
5541  Louis C.K.  United States       April 4, 2017          2017  74 min   
5794  Louis C.K.  United States  September 16, 2016          2010  84 min   
5813  Louis C.K.  United States     August 15, 2016          2015  66 min   

     duration listed_in                                        description  
5541   74 min    Movies  Louis C.K. muses on religion, eternal love, gi...  
5794   84 min    Movies  Emmy-winning comedy writer Louis C.K. brings h...  
5813   66 min    Movies  The comic puts his trademark hilarious/thought...

In [22]:
df = df.copy()
df.loc[df['show_id'] == 's5542', 'rating'] = 'PG'
df.loc[df['show_id'] == 's5795', 'rating'] = 'R'
df.loc[df['show_id'] == 's5814', 'rating'] = 'PG'

#see the rows after replace action
df[df['show_id'].isin(['s5542', 's5795', 's5814'])]

show_id   type                                 title    director  \
5541   s5542  Movie                       Louis C.K. 2017  Louis C.K.   
5794   s5795  Movie                 Louis C.K.: Hilarious  Louis C.K.   
5813   s5814  Movie  Louis C.K.: Live at the Comedy Store  Louis C.K.   

            cast        country          date_added  release_year rating  \
5541  Louis C.K.  United States       April 4, 2017          2017     PG   
5794  Louis C.K.  United States  September 16, 2016          2010      R   
5813  Louis C.K.  United States     August 15, 2016          2015     PG   

     duration listed_in                                        description  
5541   74 min    Movies  Louis C.K. muses on religion, eternal love, gi...  
5794   84 min    Movies  Emmy-winning comedy writer Louis C.K. brings h...  
5813   66 min    Movies  The comic puts his trademark hilarious/thought...

In [23]:
# Check the number of null values under rating column
df[df['rating'].isna()]

show_id     type                                              title  \
5989   s5990    Movie  13TH: A Conversation with Oprah Winfrey & Ava ...   
6827   s6828  TV Show                  Gargantia on the Verdurous Planet   
7312   s7313  TV Show                                       Little Lunch   
7537   s7538    Movie                               My Honor Was Loyalty   

             director                                               cast  \
5989              NaN                        Oprah Winfrey, Ava DuVernay   
6827              NaN  Kaito Ishikawa, Hisako Kanemoto, Ai Kayano, Ka...   
7312              NaN  Flynn Curry, Olivia Deeble, Madison Lu, Oisín ...   
7537  Alessandro Pepe  Leone Frisa, Paolo Vaccarino, Francesco Miglio...   

        country        date_added  release_year rating  duration  \
5989        NaN  January 26, 2017          2017    NaN    37 min   
6827      Japan  December 1, 2016          2013    NaN  1 Season   
7312  Australia  February 1, 2018          2015    NaN  1 Season   
7537      Italy     March 1, 2017          2015    NaN   115 min   

                                 listed_in  \
5989                                Movies   
6827  Anime Series, International TV Shows   
7312                 Kids' TV, TV Comedies   
7537                                Dramas   

                                            description  
5989  Oprah Winfrey sits down with director Ava DuVe...  
6827  After falling through a wormhole, a space-dwel...  
7312  Adopting a child's perspective, this show take...  
7537  Amid the chaos and horror of World War II, a c...

In [24]:
# Replace na values with ratings
df.loc[df['show_id'] == 's5989', 'rating'] = 'PG'
df.loc[df['show_id'] == 's6827', 'rating'] = 'TV-Y7'
df.loc[df['show_id'] == 's7312', 'rating'] = 'TV-Y7'
df.loc[df['show_id'] == 's7537', 'rating'] = 'R' 

# Display the replace nan values under rating column
df[df['show_id'].isin(['s5989','s6827','s7312','s7537'])]

show_id   type                 title         director  \
5988   s5989  Movie               13 Sins     Daniel Stamm   
6826   s6827  Movie     Gangs of New York  Martin Scorsese   
7311   s7312  Movie  Little Dragon Maiden         Hua Shan   
7536   s7537  Movie              My Ex-Ex  Nathaniel Warsh   

                                                   cast               country  \
5988  Mark Webber, Rutina Wesley, Devon Graye, Tom B...         United States   
6826  Leonardo DiCaprio, Daniel Day-Lewis, Cameron D...  United States, Italy   
7311  Leslie Cheung, Jing-Jing Yung, Chen Kuan-tai, ...             Hong Kong   
7536  André Bharti, Katherine Barrell, Ray Galletti,...                Canada   

            date_added  release_year rating duration  \
5988  January 13, 2019          2014     PG   93 min   
6826   August 20, 2019          2002  TV-Y7  167 min   
7311    August 1, 2018          1983  TV-Y7   92 min   
7536   August 15, 2017          2015      R   91 min   

                                              listed_in  \
5988                           Horror Movies, Thrillers   
6826                                             Dramas   
7311  Action & Adventure, International Movies, Roma...   
7536                          Comedies, Romantic Movies   

                                            description  
5988  A man agrees to appear on a game show with a $...  
6826  In the crime-ridden slums of New York in the 1...  
7311  Seeking to improve his combat skills, a young ...  
7536  A recently dumped attorney hires a psychic to ...

In [25]:
# Check nan after replacing duration and rating columns
df.isna().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           0
listed_in          0
description        0
dtype: int64

In [26]:
df['director'] = df['director'].fillna('Unnamed')
df['cast'] = df['cast'].fillna('Unknown')
df['country'] = df['country'].fillna(df['country'].mode()[0])
df['date_added'] = df['date_added'].fillna(df['date_added'].mode()[0])
df['rating']= df['rating'].fillna(df['rating'].mode()[0])

In [27]:
# Check nan after replacing duration and rating columns
df.isna().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
dtype: int64

### replace prefix

In [28]:
# Replace extra s
df['show_id'] = df['show_id'].str.replace('s','')

# Replace hyphen - 
df['rating'] = df['rating'].str.replace('-','')

# Display data
df 

show_id     type                  title         director  \
0          1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1          2  TV Show          Blood & Water          Unnamed   
2          3  TV Show              Ganglands  Julien Leclercq   
3          4  TV Show  Jailbirds New Orleans          Unnamed   
4          5  TV Show           Kota Factory          Unnamed   
...      ...      ...                    ...              ...   
8802    8803    Movie                 Zodiac    David Fincher   
8803    8804  TV Show            Zombie Dumb          Unnamed   
8804    8805    Movie             Zombieland  Ruben Fleischer   
8805    8806    Movie                   Zoom     Peter Hewitt   
8806    8807    Movie                 Zubaan      Mozez Singh   

                                                   cast        country  \
0                                               Unknown  United States   
1     Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2     Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...  United States   
3                                               Unknown  United States   
4     Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   
...                                                 ...            ...   
8802  Mark Ruffalo, Jake Gyllenhaal, Robert Downey J...  United States   
8803                                            Unknown  United States   
8804  Jesse Eisenberg, Woody Harrelson, Emma Stone, ...  United States   
8805  Tim Allen, Courteney Cox, Chevy Chase, Kate Ma...  United States   
8806  Vicky Kaushal, Sarah-Jane Dias, Raaghav Chanan...          India   

              date_added  release_year rating   duration  \
0     September 25, 2021          2020   PG13     90 min   
1     September 24, 2021          2021   TVMA  2 Seasons   
2     September 24, 2021          2021   TVMA   1 Season   
3     September 24, 2021          2021   TVMA   1 Season   
4     September 24, 2021          2021   TVMA  2 Seasons   
...                  ...           ...    ...        ...   
8802   November 20, 2019          2007      R    158 min   
8803        July 1, 2019          2018   TVY7  2 Seasons   
8804    November 1, 2019          2009      R     88 min   
8805    January 11, 2020          2006     PG     88 min   
8806       March 2, 2019          2015   TV14    111 min   

                                              listed_in  \
0                                         Documentaries   
1       International TV Shows, TV Dramas, TV Mysteries   
2     Crime TV Shows, International TV Shows, TV Act...   
3                                Docuseries, Reality TV   
4     International TV Shows, Romantic TV Shows, TV ...   
...                                                 ...   
8802                     Cult Movies, Dramas, Thrillers   
8803             Kids' TV, Korean TV Shows, TV Comedies   
8804                            Comedies, Horror Movies   
8805                 Children & Family Movies, Comedies   
8806     Dramas, International Movies, Music & Musicals   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

### Split string

In [29]:
# Split strings
df['cast'] = df['cast'].str.split(',').str[0]
df['listed_in'] = df['listed_in'].str.split(',').str[0]

# Display data
df

show_id     type                  title         director  \
0          1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1          2  TV Show          Blood & Water          Unnamed   
2          3  TV Show              Ganglands  Julien Leclercq   
3          4  TV Show  Jailbirds New Orleans          Unnamed   
4          5  TV Show           Kota Factory          Unnamed   
...      ...      ...                    ...              ...   
8802    8803    Movie                 Zodiac    David Fincher   
8803    8804  TV Show            Zombie Dumb          Unnamed   
8804    8805    Movie             Zombieland  Ruben Fleischer   
8805    8806    Movie                   Zoom     Peter Hewitt   
8806    8807    Movie                 Zubaan      Mozez Singh   

                 cast        country          date_added  release_year rating  \
0             Unknown  United States  September 25, 2021          2020   PG13   
1          Ama Qamata   South Africa  September 24, 2021          2021   TVMA   
2       Sami Bouajila  United States  September 24, 2021          2021   TVMA   
3             Unknown  United States  September 24, 2021          2021   TVMA   
4          Mayur More          India  September 24, 2021          2021   TVMA   
...               ...            ...                 ...           ...    ...   
8802     Mark Ruffalo  United States   November 20, 2019          2007      R   
8803          Unknown  United States        July 1, 2019          2018   TVY7   
8804  Jesse Eisenberg  United States    November 1, 2019          2009      R   
8805        Tim Allen  United States    January 11, 2020          2006     PG   
8806    Vicky Kaushal          India       March 2, 2019          2015   TV14   

       duration                 listed_in  \
0        90 min             Documentaries   
1     2 Seasons    International TV Shows   
2      1 Season            Crime TV Shows   
3      1 Season                Docuseries   
4     2 Seasons    International TV Shows   
...         ...                       ...   
8802    158 min               Cult Movies   
8803  2 Seasons                  Kids' TV   
8804     88 min                  Comedies   
8805     88 min  Children & Family Movies   
8806    111 min                    Dramas   

                                            description  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

### Rename column

In [30]:
# Rename column
df.rename(columns={"show_id":"id"}, inplace = True)
df.rename(columns={"cast":"actor"}, inplace = True)
df.rename(columns={"listed_in":"genre"}, inplace = True)
df.rename(columns={"description":"review"}, inplace = True)

# Display data
df

id     type                  title         director            actor  \
0        1    Movie   Dick Johnson Is Dead  Kirsten Johnson          Unknown   
1        2  TV Show          Blood & Water          Unnamed       Ama Qamata   
2        3  TV Show              Ganglands  Julien Leclercq    Sami Bouajila   
3        4  TV Show  Jailbirds New Orleans          Unnamed          Unknown   
4        5  TV Show           Kota Factory          Unnamed       Mayur More   
...    ...      ...                    ...              ...              ...   
8802  8803    Movie                 Zodiac    David Fincher     Mark Ruffalo   
8803  8804  TV Show            Zombie Dumb          Unnamed          Unknown   
8804  8805    Movie             Zombieland  Ruben Fleischer  Jesse Eisenberg   
8805  8806    Movie                   Zoom     Peter Hewitt        Tim Allen   
8806  8807    Movie                 Zubaan      Mozez Singh    Vicky Kaushal   

            country          date_added  release_year rating   duration  \
0     United States  September 25, 2021          2020   PG13     90 min   
1      South Africa  September 24, 2021          2021   TVMA  2 Seasons   
2     United States  September 24, 2021          2021   TVMA   1 Season   
3     United States  September 24, 2021          2021   TVMA   1 Season   
4             India  September 24, 2021          2021   TVMA  2 Seasons   
...             ...                 ...           ...    ...        ...   
8802  United States   November 20, 2019          2007      R    158 min   
8803  United States        July 1, 2019          2018   TVY7  2 Seasons   
8804  United States    November 1, 2019          2009      R     88 min   
8805  United States    January 11, 2020          2006     PG     88 min   
8806          India       March 2, 2019          2015   TV14    111 min   

                         genre  \
0                Documentaries   
1       International TV Shows   
2               Crime TV Shows   
3                   Docuseries   
4       International TV Shows   
...                        ...   
8802               Cult Movies   
8803                  Kids' TV   
8804                  Comedies   
8805  Children & Family Movies   
8806                    Dramas   

                                                 review  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]

In [31]:
# Display data again
df

id     type                  title         director            actor  \
0        1    Movie   Dick Johnson Is Dead  Kirsten Johnson          Unknown   
1        2  TV Show          Blood & Water          Unnamed       Ama Qamata   
2        3  TV Show              Ganglands  Julien Leclercq    Sami Bouajila   
3        4  TV Show  Jailbirds New Orleans          Unnamed          Unknown   
4        5  TV Show           Kota Factory          Unnamed       Mayur More   
...    ...      ...                    ...              ...              ...   
8802  8803    Movie                 Zodiac    David Fincher     Mark Ruffalo   
8803  8804  TV Show            Zombie Dumb          Unnamed          Unknown   
8804  8805    Movie             Zombieland  Ruben Fleischer  Jesse Eisenberg   
8805  8806    Movie                   Zoom     Peter Hewitt        Tim Allen   
8806  8807    Movie                 Zubaan      Mozez Singh    Vicky Kaushal   

            country          date_added  release_year rating   duration  \
0     United States  September 25, 2021          2020   PG13     90 min   
1      South Africa  September 24, 2021          2021   TVMA  2 Seasons   
2     United States  September 24, 2021          2021   TVMA   1 Season   
3     United States  September 24, 2021          2021   TVMA   1 Season   
4             India  September 24, 2021          2021   TVMA  2 Seasons   
...             ...                 ...           ...    ...        ...   
8802  United States   November 20, 2019          2007      R    158 min   
8803  United States        July 1, 2019          2018   TVY7  2 Seasons   
8804  United States    November 1, 2019          2009      R     88 min   
8805  United States    January 11, 2020          2006     PG     88 min   
8806          India       March 2, 2019          2015   TV14    111 min   

                         genre  \
0                Documentaries   
1       International TV Shows   
2               Crime TV Shows   
3                   Docuseries   
4       International TV Shows   
...                        ...   
8802               Cult Movies   
8803                  Kids' TV   
8804                  Comedies   
8805  Children & Family Movies   
8806                    Dramas   

                                                 review  
0     As her father nears the end of his life, filmm...  
1     After crossing paths at a party, a Cape Town t...  
2     To protect his family from a powerful drug lor...  
3     Feuds, flirtations and toilet talk go down amo...  
4     In a city of coaching centers known to train I...  
...                                                 ...  
8802  A political cartoonist, a crime reporter and a...  
8803  While living alone in a spooky town, a young g...  
8804  Looking to survive in a world taken over by zo...  
8805  Dragged from civilian life, a former superhero...  
8806  A scrappy but poor boy worms his way into a ty...  

[8807 rows x 12 columns]